lib/rust/parser/src/syntax/token.rs

```diff
@@ -340,6 +340,7 @@ pub struct OperatorProperties {
     is_arrow:                  bool,
     is_sequence:               bool,
     is_suspension:             bool,
+    is_autoscope:              bool,
```

Comment:
Yesterday @4e6 expressed a wish to support `!` for negation instead of `.not` method. Should that ever be done, it  could be done by adding `is_band bool` here while copying the structure of this PR, right? Or is it rather a case for macro like `if_then` method?

Reply:
It would be a new unary operator definition in the lexer and a new `OperatorProperties` field, as in this PR, but because `!` is not a special syntax (since the RHS is a general expression it is a "normal" unary operator), instead of a new `Tree` node we should expose the `OperatorProperties` field identifying it through a getter on `Token`. The latter requires a tiny refactor (currently `OperatorProperties` are blanket-excluded from serialization).

If you would like to implement it you could use a temporary `isExclamationPoint` method in `TreeToIr` to get it working and then I could add the parser changes to your PR to pass the property across languages and recognize the operator properly.